pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/utils/BlockingMultiStreamConsumer.java

```diff
@@ -240,6 +253,15 @@ protected boolean isEos(TransferableBlock element) {
       return element.isSuccessfulEndOfStreamBlock();
     }
 
+    @Override
+    protected void onConsumerFinish(TransferableBlock element) {
+      if (element.getQueryStats() != null) {
+        _stats.mergeUpstream(element.getQueryStats());
+      } else {
+        _stats.mergeUpstream(element.getSerializedStatsByStage());
```

Comment:
By doing this we don't need to deserialize the stats when they are sent using memory mailboxes. But right now we are not prioritizing sending the stats there. That is something we can improve